host/sgx/windows/enter.asm

```diff
@@ -109,71 +84,66 @@ execute_eenter:
     mov rax, ENCLU_EENTER
     ENCLU
 
-    mov ARG1OUT, rdi
-    mov ARG2OUT, rsi
+    ;; Save return values
+    mov ARG1, rdi
+    mov ARG2, rsi
 
-    ;; Restore the saved host context
-    mov rcx, HOST_CONTEXT
-    call oe_restore_host_context
+    ;; Restore flags and floating point state.
+    popfq
+    fxrstor FX_SPACE
+
+    ;; Check if an OCALL needs to be dispatched.
+    mov r10, rdi
+    shr r10, 48
+    cmp r10, OE_OCALL_CODE
+    jne return_from_ecall
 
 dispatch_ocall:
     ;; RAX = __oe_dispatch_ocall(
     ;;     RCX=arg1
     ;;     RDX=arg2
-    ;;     R8=arg1_out
-    ;;     R9=arg2_out
+    ;;     R8=&arg1
+    ;;     R9=&arg2
     ;;     [RSP+32]=TCS,
     ;;     [RSP+40]=ENCLAVE);
     ;;
     ;; Stack should already be 16-byte aligned, so only need
     ;; shadow space (32 bytes) plus stack params size (16 bytes)
     sub rsp, 48
-    mov rcx, ARG1OUT
-    mov rdx, ARG2OUT
-    lea r8, qword ptr ARG1OUT
-    lea r9, qword ptr ARG2OUT
+    mov rcx, rdi
+    mov rdx, rsi
+    lea r8, ARG1
+    lea r9, ARG2
     mov rax, qword ptr TCS
     mov qword ptr [rsp+32], rax
-    mov rax, qword ptr ENCLAVE
-    mov qword ptr [rsp+40], rax
+    mov rax, ENCLAVE
+    mov [rsp+40], rax
     call __oe_dispatch_ocall ;; RAX contains return value
```

Comment:
The comment about RAX is probably unnecessary at this point, and we should consider making `__oe_dispatch_ocall` a `void` return method since nothing actually makes use of the return value.

I would actually prefer the reverse of this change: update the linux version to invoke dispatch and let it return whether it was necessary or not, as it removes some brittle elements to maintain in assembly (e.g. OE_OCALL_CODE definitions being in sync).